### PR TITLE
Added guidelines for Integrationtests

### PR DESCRIPTION
**Pull request by paullatzelsperger**

### CODE REVIEW

File: docs/integration_testing.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+All integration tests should have the `"...IntegrationTest"` postfix to distinguish them clearly from unit tests. They
+should not reside in a separate package, because all tests should maintain package consistency to their test subject.
+
+Any credentials, secrets, passwords, etc. required by the integration tests should be passed in using environment
+variables. A good way to access them is `ConfigurationFunctions#propOrEnv()` because then the credentials can also be
+supplied via system properties.
+
+There is no one-size-fits-all guideline whether to perform setup tasks in the `@BeforeAll` or `@BeforeEach`, it will
+depend on the concrete system you're using. In most cases it is **not** advisable to deploy/provision the external
+system itself in either of them. For example, provisioning a CosmosDB or spinning up a Postgres docker container
```
Comment (ndr-brt) [inline docs/integration_testing.md:40]: So we're discouraging the use of testing frameworks like testcontainers? I used it a lot in the past and my 2 cents are that it helps a lot the execution of integration tests making them more isolated and controllable

File: docs/integration-testing.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
+# Writing Integration Tests
+
+## Definition and distinction
```
Comment (MoritzKeppler) [inline docs/integration-testing.md:3]: note that this does not match exactly other definitions, like then ones proposed in https://martinfowler.com/articles/microservice-testing/#testing-unit-introduction
Does not mean that we have to change it, just wanted to make sure that we're aware of that.

**Review by MoritzKeppler — APPROVED** [review]

**Comment by jimmarino** [comment]
@MoritzKeppler should this be merged?

**Comment by mspiekermann** [comment]
Since this one is approved I will merge this one, still keeping the revision of definitions in mind.

**Pull request merged**